### ISSUE TITLE
Confusion matrix on train dataset showing 0 in False Positives and False Negatives in Decision Tree and Random Forest models[BUG]

### ISSUE THREAD

**Issue by prafulbatra** [pycaret/pycaret#2160]
**Describe the bug**
I trained various models on my tabular dataset and plotted the confusion matrix after training. During set up I used 'test_data' parameter to provide test dataset separately. Also , for printing Confusion matrix , I used, 'use_train_data' parameter to 'True' in all models.  **In Decision Trees and Random Forest models , I am seeing zero values in both False Positives and False Negatives. And sum of points in True Positives and False Negatives is equal to total no. of data points in train set.** From the performance scores , I could see the values of Accuracy, Precision, Recall to be around 0.86 ,0.96, 0.66 respectively.In other models i.e Logistics Regression , SVM. I am getting correct values in confusion matrix on train set.

**To Reproduce**

```python
<setup1 = setup(data = data_train,test_data=data_test, target = 'PotentialFraud', session_id=100)>  
rf = create_model('rf',fold=3) 
plot_model(rf, plot = 'confusion_matrix',use_train_data=True)



```
![image](https://user-images.githubusercontent.com/68347676/153918863-4c6844b6-1b14-47fe-b22f-6571396fadfe.png)

**Expected behavior**
The values of the confusion matrix should be according to the performance scores.

**Additional context**
The values of the confusion matrix on test set is coming correct.
![image](https://user-images.githubusercontent.com/68347676/153918248-d9251f0e-b980-4eb9-bf0d-2aca2215f4d8.png)

**Versions**
<details>

2.3.6

</details>

**Comment by prafulbatra** [comment]
[code example.pdf](https://github.com/pycaret/pycaret/files/8063080/code.example.pdf)
In the above  file , I have shown in pycaret 'credit' dataset that , confusion matrix on train data is not in accordance with the performance scores of the model.

**Comment by moezali1** [comment]
@prafulbatra The scoring grid that you see is the cross-validated score. In your example, you see 3 folds and the yellow highlighted is the mean of 3 folds. 

When you pass `use_train_data=True` it actually fits the model on the entire train set and then obtains predictions on the entire train set to calculate the confusion matrix.

For tree-based models like Decision Tree and Random Forest, I would expect them to overfit on train data like you see. There are no FP and FN.

Hope this clarifies.

**Comment by prafulbatra** [comment]
@moezali1  I still didn't understand why its overfitting on RF and DT. Ideally , the values confusion matrix should be close representation and in accordance with the values of scoring grid. Shouldn't the confusion matrix on full train data be according to the model we created using create_model function ?
